smarts/core/trap_manager.py

```diff
@@ -192,19 +187,18 @@ def largest_vehicle_plane_dimension(vehicle):
                     sim, vehicle_id, agent_id, mission
                 )
             elif trap.patience_expired:
+                # Make sure there is not a vehicle in the same location
                 mission = trap.mission
-                if len(agent_vehicle_comp) > 0:
-                    agent_vehicle_comp.sort(
-                        key=lambda v: squared_dist(v[0], mission.start.position)
-                    )
-
-                    # Make sure there is not an agent vehicle in the same location
-                    pos, largest_dimension, _ = agent_vehicle_comp[0]
+                overlapping = False
+                for pos, largest_dimension, _ in vehicle_comp:
                     if squared_dist(pos, mission.start.position) < largest_dimension:
```

Comment:
Truthfully, this should be more like:
```suggestion
                    if squared_dist(pos, mission.start.position) < (largest_dimension+<new_vehicle_dimensions>) ** 2:
```

Reply:
What should be that dimension? The length of the new vehicle?

Reply:
Yeah, which isn't really completely specified at this point (since this is where we _decide_ whether to add it).  However, I just pushed an update which uses something similar to what's done in `build_agent_vehicle()` in `vehicle.py` to do this.